core/src/main/scala/org/bitcoins/core/wallet/utxo/ConditionalPath.scala

```diff
@@ -0,0 +1,50 @@
+package org.bitcoins.core.wallet.utxo
+
+/** Represents the spending branch being taken in a ScriptPubKey's execution
+  *
+  * If you over-specify a path, such as giving a condition where none is needed,
+  * then the remaining over-specified path will be ignored.
+  *
+  * For example, if you wanted to spend a ConditionalScriptPubKey(P2PK1, P2PK2)
+  * (which looks like OP_IF <P2PK1> OP_ELSE <P2PK2> OP_ENDIF) with the P2PK1 case,
+  * then you would construct a ConditionalSpendingInfo using nonNestedTrue as your
+  * ConditionalPath. Otherwise if you wanted to use P2PK2 you would use nonNestedFalse.
+  */
+sealed trait ConditionalPath {
+  def headOption: Option[Boolean]
```

Comment:
Let's move this into a companion object if you feel like this is necessary (similar to `scriptWitnessOpt`, `receemScriptOpt`, same argument, `NoCondition.headOption()` represents an illegal state).

Reply:
No illegal states here, even the standard library has `Nil.headOption`